irohad/synchronizer/impl/synchronizer_impl.cpp

```diff
@@ -88,28 +87,21 @@ namespace iroha {
           if (not storage) {
             return;
           }
-          auto chain =
-              blockLoader_->retrieveBlocks(shared_model::crypto::PublicKey(
-                  {signature.pubkey.begin(), signature.pubkey.end()}));
+          auto chain = blockLoader_->retrieveBlocks(
+              shared_model::crypto::PublicKey(signature->publicKey()));
           if (validator_->validateChain(chain, *storage)) {
             // Peer send valid chain
             mutableFactory_->commit(std::move(storage));
-
-            // TODO: 07-03-2018 Alexey Chernyshov remove this after relocation to
-            // shared_model https://soramitsu.atlassian.net/browse/IR-903
-            notifier_.get_subscriber().on_next(chain.map([](auto block) {
-              std::unique_ptr<iroha::model::Block> old_block(
-                  block->makeOldModel());
-              return *old_block;
-            }));
+            notifier_.get_subscriber().on_next(
+                chain.map([](auto block) { return block; }));
```

Comment:
is it possible to pass block by const ref?

Reply:
again, we should be able to remove map